tests/audit_sync

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+log_file="$(grep -E '^\s*log_file\s*=\s*' /etc/audit/auditd.conf | \
+	sed -e 's/^.*=[[:space:]]*//')"
+log_file="${log_file:-/var/log/audit/audit.log}"
+
+key="audit-sync=$(head -c 10 /dev/urandom | hexdump -e '"%02x"')"
+
+# generate an event and wait for it to appear in the log
+tail -s 0.05 -n +1 -F "$log_file" | \
+	timeout 10s sed -e "/^type=USER.*msg='$key/Q" >/dev/null &
+
+auditctl -m "$key"
+
+wait $!
```

Comment:
I'm not in love with the idea that this is a separate script that lives outside the individual tests for reasons already discussed in the PR comments (self contained tests).

This also seems overly complex, why couldn't we do something like this in the tests?

```
# 
# untested perl-ish pseudo code, probably buggy/flawed as a result
#

system("auditctl -m <somerandomstring>");
while (system("ausearch -m USER | grep -q msg=<string>")) {
     # sleep and/or sync, possibly a loop counter and a fatal error to prevent excessive looping
}
```

Reply:
For performance reasons - every time you call ausearch, it goes through the whole audit log (including rotated logs). I tried to minimize the number of times the logs need to be traversed. Also, the script doesn't use the somewhat slower ausearch (which parses each record), but ony a simple grep.

Reply:
At this point in time I'm not overly concerned about the overhead of an additional ausearch call for each test; the test suite runs rather quickly as it currently stands.  If you are bothered by the ausearch call you could easily substitute a call to sed/grep.

Reply:
I want to add, the important part about this comment thread is not the tool that is used to search through the logs for the marker, but rather the idea that this explicit check (vs an implicit sleep) is incorporated into the tests and not a separate script.